src/visuals/sprite.rs

```diff
@@ -399,15 +440,33 @@ impl Sprite {
     ///
     /// **Warning**: this method is not meant to be used
     pub fn to_str(&self) -> String {
-        format!(
-            "Sprite,{},{},\"{}\",{},{}\n{}",
-            self.layer,
-            self.origin,
-            self.path,
-            self.pos.x,
-            self.pos.y,
-            self.events.to_str()
-        )
+        match &self.animation {
+            Some(animation) => {
+                return format!(
+                    "Animation,{},{},\"{}\",{},{},{},{},{}\n{}",
```

Comment:
you can just do the following:

```
format!(
    "{},...",
    match &self.animation {
        Some(animation) => "Animation",
        None => "Sprite",
    },
    ...
)
````

Reply:
Wouldn't this cause some issues because the parameters count is different? aka we have 3 more args at the end of the Element line ``frame_count,frame_delay,loop_type`` how would you handle this?

Reply:
I wrote this before telling about changing the struct to enum, but of could you would have to do that if you were to use `enum`s instead.

You can also use the `..` tag so it would be, in the best case:

```
match &self.sprite_type {
    SpriteType::Sprite => "Sprite",
    SpriteType::Animation{..} => "Animation",
}
```

Reply:
Discussed with @S-Traut, since there are new values and the line is formatted differently depending on whether it is a `Sprite` or `Animation`, it is clearer to keep the current `match`.